src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_gligen.py

```diff
@@ -0,0 +1,827 @@
+
+        device = self._execution_device
+        # here `guidance_scale` is defined analog to the guidance weight `w` of equation (2)
+        # of the Imagen paper: https://arxiv.org/pdf/2205.11487.pdf . `guidance_scale = 1`
+        # corresponds to doing no classifier free guidance.
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        # 3. Encode input prompt
+        prompt_embeds = self._encode_prompt(
+            prompt,
+            device,
+            num_images_per_prompt,
+            do_classifier_free_guidance,
+            negative_prompt,
+            prompt_embeds=prompt_embeds,
+            negative_prompt_embeds=negative_prompt_embeds,
+        )
+
+        # 4. Prepare timesteps
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        # 5. Prepare latent variables
+        num_channels_latents = self.unet.in_channels
+        latents = self.prepare_latents(
+            batch_size * num_images_per_prompt,
+            num_channels_latents,
+            height,
+            width,
+            prompt_embeds.dtype,
+            device,
+            generator,
+            latents,
+        )
+
+        # 5.1 Prepare GLIGEN variables
+        if gligen_phrases is not None:
+            assert len(gligen_phrases) == len(gligen_boxes)
+            max_objs = 30
+            if len(gligen_boxes) > max_objs:
+                warnings.warn(
+                    f"More that {max_objs} objects found. Only first {max_objs} objects will be processed.",
+                    FutureWarning,
+                )
+                gligen_phrases = gligen_phrases[:max_objs]
+                gligen_boxes = gligen_boxes[:max_objs]
+            _boxes = gligen_boxes
+            # prepare batched input to the PositionNet (boxes, phrases, mask)
+            # Get tokens for phrases from pre-trained CLIPTokenizer
+            tokenizer_inputs = self.tokenizer(gligen_phrases, padding=True, return_tensors="pt").to(device)
+            # For the token, we use the same pre-trained text encoder
+            # to obtain its text feature
+            _text_embeddings = self.text_encoder(**tokenizer_inputs).pooler_output
+            n_objs = min(len(_boxes), max_objs)
+            dtype = self.text_encoder.dtype
+            # For each entity, described in phrases, is denoted with a bounding box,
+            # we represent the location information as (xmin,ymin,xmax,ymax)
+            boxes = torch.zeros(max_objs, 4, device=device, dtype=dtype)
+            boxes[:n_objs] = torch.tensor(_boxes[:n_objs])
+            text_embeddings = torch.zeros(max_objs, self.unet.cross_attention_dim, device=device, dtype=dtype)
+            text_embeddings[:n_objs] = _text_embeddings[:n_objs]
+            # Generate a mask for each object that is entity described by phrases
+            masks = torch.zeros(max_objs, device=device, dtype=dtype)
+            masks[:n_objs] = 1
```

Comment:
Changed `_text_embedding`, `gligen_boxes` to have length `n_objs`.

Do boxes, text_embeddings has to be a fixes size max_obj?  yes, that's how GLIGEN authors intended it to be. It would be good to keep it that way for now